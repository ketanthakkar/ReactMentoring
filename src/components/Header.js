import React from 'react';
import Title from './Title';
import Search from './Search';

const Header = () => (
        <header className="header-section">
            <Title />
            <Search />
        </header>
)

export default Header;