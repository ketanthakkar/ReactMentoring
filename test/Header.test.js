import React from 'react';
import { mount } from 'enzyme';
import Header from "../src/components/Header";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({});

describe('should render Header correctly', () => {
  const tree = mount(
    <Provider store={store}>
      <Header />
    </Provider>
    );
  test('Snapshot test', () => {
    expect(tree).toMatchSnapshot();
  });
});