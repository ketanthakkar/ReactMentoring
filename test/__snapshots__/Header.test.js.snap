// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly Snapshot test 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Header>
    <header
      className="header-section"
    >
      <Title>
        <div
          className="title-section"
        >
          <h4
            className="netflix-text"
          >
            netflixroulette
          </h4>
        </div>
      </Title>
      <Connect(Search)>
        <Search
          fetchMovies={[Function]}
          filterMovies={[Function]}
          filterMoviesBy={[Function]}
        >
          <section
            className="search-section"
          >
            <h4
              className="white-text"
            >
              FIND YOUR
            </h4>
            <input
              className="white-text search-title"
              onChange={[Function]}
              placeholder="Enter title"
              type="text"
            />
            <div
              className="search-container"
            >
              <div
                className="search-selection"
              >
                <label
                  className="white-text"
                >
                  SEARCH BY
                </label>
                <button
                  className="white-text"
                  id="title-btn"
                  onClick={[Function]}
                >
                  TITLE
                </button>
                <button
                  className="white-text nonselected-color"
                  id="genre-btn"
                  onClick={[Function]}
                >
                  GENRE
                </button>
              </div>
              <button
                className="white-text"
                onClick={[Function]}
              >
                SEARCH
              </button>
            </div>
          </section>
        </Search>
      </Connect(Search)>
    </header>
  </Header>
</Provider>
`;
